src/components/Comments.jsx

```diff
@@ -16,35 +20,82 @@ export default function Comments() {
       .catch((err) => {
         console.log(err);
       });
-  }, []);
+  }, [articleId]);
+
+  const handleCommentSubmit = (e) => {
+    e.preventDefault();
+    if (newComment.trim() === "") {
+      setSubmitErr("Comment is empty...");
+      return;
+    }
+    setIsSubmit(true);
+
+    const comment = {
+      username: "tickle122",
+      body: newComment,
+    };
+    baseApi
+      .post(`/articles/${articleId}/comments`, comment)
+      .then((res) => {
+        setNewComment("");
+        setSubmitErr(null);
+        setComments((showComments) => [res.data.comment, ...showComments]);
+        setIsSubmit(false);
+      })
+      .catch((err) => {
+        setSubmitErr("Sending comment failed. Please try again!", err);
+        setIsSubmit(false);
+      });
+  };
+
   if (isLoading) {
     return <h1>Loading</h1>;
   }
+
   return (
-    <div>
-      {comments.length === 0 ? (
-        <p>No comments</p>
-      ) : (
-        <ul>
-          {comments.map((comment) => (
-            <li
-              key="comment.comment_id"
-              style={{
-                border: "1px solid white",
-                // marginBottom: "10px",
-                padding: "10px",
-              }}
-            >
-              <p>
-                <strong>Name: {comment.author}</strong>
-              </p>
-              <p>{comment.body}</p>
-              <p>Votes: {comment.votes}</p>
-              <p>Date: {comment.created_at}</p>
-            </li>
-          ))}
-        </ul>
-      )}
-    </div>
+    <>
+      <div>
+        {comments.length === 0 ? (
+          <p>No comments</p>
+        ) : (
+          <ul>
+            {comments.map((comment) => (
+              <li
+                key={comment.comment_id}
+                style={{
+                  border: "1px solid white",
+                  // marginBottom: "10px",
+                  padding: "10px",
+                }}
+              >
+                <p>
+                  <strong>Name: {comment.author}</strong>
+                </p>
+                <p>{comment.body}</p>
+                <p>Votes: {comment.votes}</p>
+                <p>Date: {comment.created_at}</p>
+              </li>
+            ))}
+          </ul>
+        )}
+      </div>
+
+      <div>
+        <h3>Add a Comment</h3>
+        <form onSubmit={handleCommentSubmit}>
+          <textarea
+            style={{ height: "150px", width: "100%" }}
+            value={newComment}
+            placeholder="Write a new Comment..."
+            onChange={(e) => setNewComment(e.target.value)}
+            disabled={isSubmit}
+          ></textarea>
+          <button type="submit" style={{ backgroundColor: "grey" }}>
+            {isSubmit ? "Comment sent successfully!" : "Send Comment"}
+          </button>
+          {submitErr && <p style={{ color: "red" }}>{submitErr}</p>}
```

Comment:
When I slow the internet speed down I noticed that the button changes to say "Comment sent successfully" even before the request completes, you might have to check over this logic